### PR TITLE
fix: qualify Rails as ::Rails to avoid namespace clashes with other gems

### PR DESCRIPTION
**Pull request by SingleShot**
### All Submissions:

- [ ] I've checked to ensure there aren't other open [Pull Requests](https://github.com/cerebris/jsonapi-resources/pulls) for the same update/change.
- [x] I've submitted a [ticket](https://github.com/cerebris/jsonapi-resources/issues) for my issue if one did not already exist.
- [x] My submission passes all tests. (Please run the full test suite locally to cut down on noise from travis failures.)
- [x] I've used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message or the description.
- [ ] I've added/updated tests for this change.

### New Feature Submissions:

- [ ] I've submitted an issue that describes this feature, and received the go ahead from the maintainers.
- [ ] My submission includes new tests.
- [x] My submission maintains compliance with [JSON:API](http://jsonapi.org/).

### Bug fixes and Changes to Core Features:

- [x] I've included an explanation of what the changes do and why I'd like you to include them.
- [ ] I've provided test(s) that fails without the change.

### Test Plan:
I was unable to get the bug template to run and could not find instructions on how to do so.

### Reviewer Checklist:
- [ ] Maintains compliance with JSON:API
- [ ] Adequate test coverage exists to prevent regressions

### CODE REVIEW

**Comment by lgebhardt** [comment]
Thanks @SingleShot! What version of JR are you using. Right now we have multiple versions of the gem backed by different branches. The master branch is currently being used for development of the new v0.11 branch, though the current beta of that is on the v0-11-dev branch which will eventually merge to master.

I can take this change to the version you are using and get a release out for you.

**Comment by SingleShot** [comment]
Hi. I tried integrating both `0.10.7` and `master` to the same effect. I'm happy to work with the 11 beta.

File: lib/jsonapi-resources.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@
 require 'jsonapi/cached_response_fragment'
 require 'jsonapi/response_document'
 require 'jsonapi/acts_as_resource_controller'
-if Rails::VERSION::MAJOR >= 6
+if ::Rails::VERSION::MAJOR >= 6
```
Comment (bf4) [inline lib/jsonapi-resources.rb:12]: Did this issue come up in response to an experience failure? This specific changes and others in the PR shouldn't be needed, I wouldn't think.

Comment (SingleShot) [reply lib/jsonapi-resources.rb:12]: I thought my issue was linked here but I'm not finding it. [Here](https://github.com/cerebris/jsonapi-resources/issues/1413) it is. Basically there is a namespace clash when using this project and a different JSON:API project together.

**Comment by lgebhardt** [comment]
I added this in #1421 and it will go into the master branch when we merge in the v0.11 dev branch

**Pull request closed without merging**